supply/spec/uploader_spec.rb

```diff
@@ -453,5 +453,152 @@ def ==(other)
+        allow(client).to receive(:tracks).with('alpha').and_return([track])
+        allow(release).to receive(:version_codes).and_return([version_code])
+        allow(client).to receive(:update_track)
+      end
+
+      shared_examples 'updates track with correct status and rollout' do |rollout:, release_status:, expected_status:, expected_user_fraction:|
+        before do
+          release.status = Supply::ReleaseStatus::IN_PROGRESS
+          track.releases = [release]
+          config[:rollout] = rollout
+          config[:release_status] = release_status
+        end
+
+        it "sets status to #{expected_status} and user_fraction to #{expected_user_fraction.inspect}" do
+          expect(client).to receive(:update_track).with('alpha', track) do |_, track_param|
+            expect(track_param.releases).to eq([release])
+            expect(track_param.releases.first.user_fraction).to eq(expected_user_fraction)
+            expect(track_param.releases.first.status).to eq(expected_status)
+          end
+          subject.update_rollout
+        end
+      end
+
+      shared_examples 'raises error for invalid rollout' do |rollout:, release_status:, expected_error:|
+        before do
+          release.status = Supply::ReleaseStatus::IN_PROGRESS
+          track.releases = [release]
+          config[:rollout] = rollout
+          config[:release_status] = release_status
+        end
+
+        it 'raises an error' do
+          expect { subject.update_rollout }.to raise_error(expected_error)
+        end
+      end
+
+      context 'when rollout is nil' do
+        context 'when release_status is DRAFT' do
+          include_examples 'updates track with correct status and rollout',
+            rollout: nil,
+            release_status: Supply::ReleaseStatus::DRAFT,
+            expected_status: Supply::ReleaseStatus::DRAFT,
+            expected_user_fraction: nil
+        end
+
+        context 'when release_status is IN_PROGRESS' do
+          include_examples 'raises error for invalid rollout',
+            rollout: nil,
+            release_status: Supply::ReleaseStatus::IN_PROGRESS,
+            expected_error: /You need to provide a rollout value when release_status is set to 'inProgress'/
+        end
+
+        context 'when release_status is COMPLETED' do
+          include_examples 'updates track with correct status and rollout',
+            rollout: nil,
+            release_status: Supply::ReleaseStatus::COMPLETED,
+            expected_status: Supply::ReleaseStatus::COMPLETED,
+            expected_user_fraction: nil
+        end
+      end
+
+      context 'when rollout is 0.5' do
+        context 'when release_status is DRAFT' do
+          include_examples 'updates track with correct status and rollout',
+            rollout: 0.5,
+            release_status: Supply::ReleaseStatus::DRAFT,
+            expected_status: Supply::ReleaseStatus::DRAFT,
+            expected_user_fraction: 0.5
+        end
```

Comment:
Tbh I'm actually not sure this is valid. Does Google Play allow you to set a rollout value to a release that is in Draft state? (e.g. so that when you turn that draft into an actual release by setting its status to `inProgress` later, it'd use that rollout value previously set)?

if anyone knows, or has some opportunity to test with a real-life example, I'm interested.